### PR TITLE
DDF-3808 Append queryId to metacard action provider urls

### PR DESCRIPTION
**Pull request by gordocanchola**
Passing query Id to Metarcard actions and appending it to url
Setting queryId on action in QueryResponse
Setting queryId on action in QueryResult
Updating package.json to add querystring to dependencies

#### What does this PR do?
This PR appends queryId as a query param to the urls generated by the metacard action providers.

#### Who is reviewing it? 
(please choose AT LEAST two reviewers that need to approve the PR before it can get merged)
@djblue @AzGoalie @GabrielFabian @mojogitoverhere @dcruver @BernardIgiri @rymach @pklinef 
#### Select relevant component teams: 
https://github.com/orgs/codice/teams/ui
#### Choose 2 committers to review/merge the PR. 
(please choose ONLY two committers from below, delete the rest)
@andrewkfiedler
@jlcsmith 

#### How should this be tested? (List steps with links to updated documentation)
Using Intrigue, perform a search
Open Inspector for a selected result
Select Actions
Hover cursor over metacard actions or Right click -> Copy link address
Verify there is a query param for query id only for actions with the queryId parameter

#### What are the relevant tickets?
[DDF-3808](https://codice.atlassian.net/browse/DDF-3808)
#### Checklist:
- [ ] Documentation Updated
- [ ] Update / Add Unit Tests
- [ ] Update / Add Integration Tests

#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.

### CODE REVIEW

File: ui/packages/catalog-ui-search/package.json [hunk 1]
```diff
@@ -60,6 +60,7 @@
     "openlayers": "4.6.5",
     "plotly.js": "1.10.5",
     "q": "1.5.0",
+    "querystring": "^0.2.0",
```
Comment (jlcsmith) [inline ui/packages/catalog-ui-search/package.json]: Is the ^ a typo?

Comment (BernardIgiri) [reply ui/packages/catalog-ui-search/package.json]: No, that's npm. It means that version or greater: https://docs.npmjs.com/misc/semver#caret-ranges-123-025-004

Comment (gordocanchola) [reply ui/packages/catalog-ui-search/package.json]: I did an `npm install querystring` and it added that line for me.

Comment (mojogitoverhere) [reply ui/packages/catalog-ui-search/package.json]: I think some linter complains about using version ranges like that

Comment (gordocanchola) [reply ui/packages/catalog-ui-search/package.json]: OK. I can remove it.

**Comment by jlcsmith** [comment]
Verified queryId added to all metacard action URLs

**Comment by jlcsmith** [comment]
build now

File: ui/packages/catalog-ui-search/src/main/webapp/js/model/MetacardAction.js [hunk 2]
```diff
@@ -11,17 +11,32 @@
...
         };
     },
     getExportType: function() {
         return this.get('title').replace(/^Export( as)?\s+\b/, '');
+    },
+    initialize: function() {
+        this.handleQueryId();
+        this.listenTo(this, 'change:queryId', this.handleQueryId);
+    },
+    handleQueryId: function() {
```
Comment (jlcsmith) [inline ui/packages/catalog-ui-search/src/main/webapp/js/model/MetacardAction.js]: ❓ I wonder if there is a way to only add the query ID if the action requires it.  Thoughts?

File: ui/yarn.lock [hunk 3]
```diff
@@ -11731,7 +11731,7 @@ querystring-es3@^0.2.0, querystring-es3@~0.2.0:
   version "0.2.1"
   resolved "https://registry.yarnpkg.com/querystring-es3/-/querystring-es3-0.2.1.tgz#9ec61f79049875707d69414596fd907a4d711e73"
 
-querystring@0.2.0:
+querystring@0.2.0, querystring@^0.2.0:
```
Comment (mojogitoverhere) [inline ui/yarn.lock]: You might need to repush this yarn.lock file. I suggest doing a new `yarn install` and see if it changes. I don't think `querystring@^0.2.0` should be there after the `^` was removed from the package.json file.

File: ui/packages/catalog-ui-search/src/main/webapp/js/model/MetacardAction.js [hunk 4]
```diff
@@ -11,17 +11,32 @@
...
+        if (this.get('queryId') !== undefined) {
+            const parsedU = URL.parse(this.get('url'));
+            const parsedQs = QS.parse(parsedU.query);
+            parsedQs.queryId = this.get('queryId');
+            this.set('url', `${parsedU.protocol}//${parsedU.host}${parsedU.pathname}?${QS.stringify(parsedQs)}`);
```
Comment (mojogitoverhere) [inline ui/packages/catalog-ui-search/src/main/webapp/js/model/MetacardAction.js]: We might need to create the url string without the template literal, looks like they aren't supported in IE11 https://caniuse.com/#feat=template-literals. I would double check with @andrewkfiedler and @djblue though.

**Comment by jlcsmith** [comment]
build now

File: ui/packages/catalog-ui-search/src/main/webapp/js/model/MetacardAction.js (hunk 2 above)
Comment (gordocanchola) [reply ui/packages/catalog-ui-search/src/main/webapp/js/model/MetacardAction.js]: Updated to only add the query id if the action requires it.

**Pull request merged**